sys/fmt/fmt.c

```diff
@@ -508,22 +508,16 @@ uint32_t scn_u32_hex(const char *str, size_t n)
 
 void print(const char *s, size_t n)
 {
+    if (IS_USED(MODULE_STDIO_NATIVE)) {
+        /* native gets special treatment as native's stdio code is ...
+         * special */
+        printf("%.*s", (int)n, s);
```

Comment:
I previously used `%*s` instead of `%.*s` :/ With the `.` forgotten, it gives the "minimum field width" and `tests/fmt_print` only passed because the strings happend to be zero terminated. I spiced up the test to detect just this issue by printing `Test not passed` instead of `Test passed` when `print()` is printing longer than the given length.

May I squash?

Reply:
Sure, please squash